service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/AbstractEmailContentProvider.java

```diff
@@ -50,19 +45,72 @@ protected AbstractEmailContentProvider(String uiBaseUrl) {
             timeFramePresent = "Yes";
         }
 
-        if (orderType.isEmpty()) {
+        if (isDataPresent) {
             dataPresent = "No";
             fullStop = "Yes";
         }
 
-        return ImmutableMap.<String, String>builder()
-            .putAll(orderTypeArray.build())
+        return ImmutableMap.<String, Object>builder()
+            .put("ordersAndDirections", ordersAndDirectionsList)
             .put("dataPresent", dataPresent)
             .put("fullStop", fullStop)
             .put("timeFramePresent", timeFramePresent)
             .put("timeFrameValue", Optional.ofNullable((String) hearing.get("timeFrame")).orElse(""))
-            .put(DIRECTIONS_KEY, !directions.isEmpty() ? "^" + directions : "")
             .put("reference", String.valueOf(caseDetails.getId()))
             .put("caseUrl", uiBaseUrl + "/case/" + JURISDICTION + "/" + CASE_TYPE + "/" + caseDetails.getId());
     }
+
+    private boolean buildOrders(Map orders, ImmutableSet.Builder<String> setBuilder) {
+        List orderType = (List) Optional.ofNullable(orders.get("orderType")).orElse(ImmutableList.builder().build());
+        List emergencyProtectionOrders = (List) Optional.ofNullable(orders.get("emergencyProtectionOrders"))
+            .orElse(ImmutableList.builder().build());
+
+        for (int i = 0; i < orderType.size(); i++) {
+            if (orderType.get(i) != "OTHER") {
```

Comment:
Use enum you created e.g. `OrderType.OTHER.name()`. This way it will be less fragile for refactoring.

Reply:
```
for (int i = 0; i < orderType.size(); i++) {
    OrderType value = OrderType.valueOf((String) orderType.get(i))
    if (value != OrderType.OTHER) {
        setBuilder.add(value.getLabel());
    }
}
```

☝️ gives you nice ability to have type safe enum comparison.